lib/src/sql/latlng.rs

```diff
@@ -0,0 +1,226 @@
+use crate::sql::comment::mightbespace;
+use crate::sql::error::{Error, IResult};
+use crate::sql::Operator::Dec;
+use geo::Point;
+use nom::branch::alt;
+use nom::character::streaming::char;
+use nom::combinator::map;
+use nom::complete::tag;
+use nom::error::ParseError;
+use nom::number::complete::double;
+use nom::sequence::tuple;
+use nom::Parser;
+use std::fmt;
+use std::fmt::Formatter;
+use std::ops::Mul;
+use std::rc::Rc;
+
+/// Indicates one of four directions used for geographic orientation.
+/// North and South are vertical, East and West are horizontal.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum CardinalDirection {
+	North,
+	South,
+	East,
+	West,
+}
+
+/// In terms of latitude and longitude, the following rules apply:
+/// - North and East are positive
+/// - South and West are negative
+///
+/// This implementation allows callers to multiply a floating point
+/// number by an associated [`CardinalDirection`] in order to produce
+/// a signed value representing the correct latitude or longitude.
+impl Mul<CardinalDirection> for f64 {
+	type Output = f64;
+
+	fn mul(self, rhs: CardinalDirection) -> Self::Output {
+		match rhs {
+			CardinalDirection::North => self,
+			CardinalDirection::South => self * -1.0,
+			CardinalDirection::East => self,
+			CardinalDirection::West => self * -1.0,
+		}
+	}
+}
+
+impl fmt::Display for CardinalDirection {
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		match self {
+			CardinalDirection::North => write!(f, "N"),
+			CardinalDirection::South => write!(f, "S"),
+			CardinalDirection::East => write!(f, "E"),
+			CardinalDirection::West => write!(f, "W"),
+		}
+	}
+}
+
+/// Represents a degree of latitude or longitude and an associated
+/// [`CardinalDirection`]. For example "N 40.446°" and "W 79.982°"
+/// are both representable by this type.
+#[derive(Debug, Clone, Copy, PartialEq)]
+struct CardinalDegree(CardinalDirection, f64);
+
+impl CardinalDegree {
+	fn value(&self) -> f64 {
+		self.1
+	}
+
+	fn direction(&self) -> CardinalDirection {
+		self.0
+	}
+}
+
+impl fmt::Display for CardinalDegree {
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		// todo: precision?
+		write!(f, "{:.2}°{}", self.1, self.0)
+	}
+}
```

Comment:
@rushmorem thoughts?

Reply:
This is not my area. Perhaps @tobiemh or @finnbear might be able to help.

Reply:
Yeah, that's a good point. These are always translated into `Geometry::Point` so there's no need to display them as latlng. I believe there are some tests that use this to check that things are being parsed correctly, but I could certainly remove them since there are more comprehensive tests covering parsing.

Reply:
If you did keep the tests, you could move the display logic (with any precision you like) into a private function used by those tests only :wink:

Reply:
@finnbear Done!